package_manager/dpkg_parser.py

```diff
@@ -72,59 +75,46 @@ def main():
                               args.packages_gz_url, args.package_prefix)
         util.build_os_release_tar(args.distro, OS_RELEASE_FILE_NAME, OS_RELEASE_PATH, OS_RELEASE_TAR_FILE_NAME)
     else:
-        download_dpkg(args.package_files, args.packages, args.workspace_name)
+        download_dpkg(args.package_files, args.packages, args.workspace_name, args.versionsfile)
 
-def download_dpkg(package_files, packages, workspace_name):
+def download_dpkg(package_files, packages, workspace_name, versionsfile):
     """ Using an unzipped, json package file with full urls,
      downloads a .deb package
 
     Uses the 'Filename' key to download the .deb package
     """
-    pkg_vals_to_package_file_and_sha256 = {}
     package_to_rule_map = {}
     package_to_version_map = {}
     package_file_to_metadata = {}
-    for pkg_vals in set(packages.split(",")):
-        pkg_split = pkg_vals.split("=")
-        if len(pkg_split) != 2:
-            pkg_name = pkg_vals
-            pkg_version = ""
-        else:
-            pkg_name, pkg_version = pkg_split
+    for pkg_name in set(packages.split(",")):
+        pkg = {}
         for package_file in package_files.split(","):
             if package_file not in package_file_to_metadata:
```

Comment:
I'm now wondering if this check as well as `package_file_to_metadata` are necessary. (I don't remember what values `package_files` hold though.) Can't we just always open and read the file, and then just get `metadata[pkg_name]`? It is really possible that `package_files` may have duplicate items and the optimization with this check is worth?

Reply:
the check is there because of the two nested loops:
 1. first loop over all packages that we want to download
 2. loop over all repository url metadata to see if it has a version. If not available in the map, download it first
      2.1 **Added**: check if metadata found is more recent that other previous package metadata from another package_file

If you want; we can optimize it to fill the package_file_to_metadata before looping over all the packages. In that case the check you pointed out can be removed.

Reply:
Ah, I think I get it. This check is worth because (probably) `package_file` is a remote URL for a JSON file hence opening and reading it is expensive? If so, I think this check makes sense.

Reply:
Correct, that's why I left it in the place it was originally written (works like a cache).